src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -369,21 +433,10 @@ void DistributedTreeImpl<DeviceType>::reassessStrategy(
           farthest_distances(i) = max(farthest_distances(i), distances(j));
       });
 
-  // Identify what ranks may have leaves that are within that distance.
-  Kokkos::View<decltype(intersects(Sphere{})) *, DeviceType> radius_searches(
-      Kokkos::view_alloc(
-          space, Kokkos::WithoutInitializing,
-          "ArborX::DistributedTree::query::reassessStrategy::queries"),
-      n_queries);
-  Kokkos::parallel_for(
-      "ArborX::DistributedTree::query::bottom_trees_within_that_distance",
-      Kokkos::RangePolicy<ExecutionSpace>(space, 0, n_queries),
-      KOKKOS_LAMBDA(int i) {
-        radius_searches(i) = intersects(Sphere{
-            getGeometry(Access::get(queries, i)), farthest_distances(i)});
-      });
-
-  query(top_tree, space, radius_searches, indices, offset);
+  query(top_tree, space,
+        WithinDistanceFromPredicates<Predicates, decltype(farthest_distances)>{
```

Comment:
`decltype(farthest_distances)` is always a float.  There is no point in having this template parameter.

Reply:
Fartherst distances is a view.